ledger/complete/ledger.go

```diff
@@ -200,20 +262,31 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, trieUpdate *
 
 	l.metrics.UpdateCount()
 
-	walChan := make(chan error)
+	walChan := make(chan walUpdateResult)
 
 	go func() {
-		walChan <- l.wal.RecordUpdate(trieUpdate)
+		segmentNum, err := l.wal.RecordUpdate(trieUpdate)
```

Comment:
Writing to WAL files happens concurrently, we need a way to ensure the order of trie updates written to the file and the updates pushed to trieUpdateCh is consistent. The current implementation can not guarantee that

Reply:
> The current implementation can not guarantee that 

The current implementation is not PR 2770 because it was superceded by PR #2792 days ago.

In PR #2792, when `Compactor` receives `trieUpdate` from channel, `Compactor` 
- writes update in WAL, 
- signals to `ledger.Set` when WAL update is completed, 
- waits for trie update completion (adding new trie ledger state)

Because all these step, the order of trie updates and WAL updates are consistent.